yt_dlp/extractor/tumblr.py

```diff
@@ -232,13 +216,160 @@ class TumblrIE(InfoExtractor):
+        'info_dict': {
+            'id': 'x8hczf5',
+            'ext': 'mp4',
+            'title': 'Trying a viral pizza bagel in NYC',
+            'description': 'md5:17e52b32ae21f23940912b3efff17b74',
+            'duration': 58,
+            'uploader': 'Insider',
+            'uploader_id': 'x29n239',
+            'tags': ['food', 'video', 'pizza', 'foody', 'bagel', 'bagels', 'feed:ins', 'feed:live', 'jacky barile', 'reels'],
+            'timestamp': 1674064861,
+            'upload_date': '20230118',
+            'age_limit': 0,
+            'thumbnail': 're:https?://.+',
+            'view_count': int,
+            'like_count': int,
+        },
+    }, {
+        'note': 'tiktok video embed',
+        'url': 'https://selfisekai.tumblr.com/post/706885498468270080',
+        'info_dict': {
+            'id': '7098761136534867205',
+            'ext': 'mp4',
+            'title': 'md5:6e62de3b0157d2ed9999c7d61c6485f2',
+            'description': 'md5:6e62de3b0157d2ed9999c7d61c6485f2',
+            'creator': 'Poznań 🐐',
+            'uploader': 'miasto_poznan',
+            'uploader_id': '6998015174997607430',
+            'uploader_url': 'https://www.tiktok.com/@MS4wLjABAAAAzgdq4BFpzWfUWq4sO84_HGHHdS7cItoTDdAscuSxYIUQKJvZZ9j99wPe0RuqJpaR',
+            'duration': 46,
+            'timestamp': 1652809127,
+            'upload_date': '20220517',
+            'thumbnail': r're:https?://[^/]+\.tiktokcdn\.com/.+',
+            'artist': 'The King Khan & BBQ Show',
+            'album': 'Love You So',
+            'track': 'Love You So',
+            'view_count': int,
+            'like_count': int,
+            'comment_count': int,
+            'repost_count': int,
+        },
+    }, {
+        'note': 'tumblr video AND youtube embed',
+        'url': 'https://selfisekai.tumblr.com/post/706895394042511360',
+        'info_dict': {
+            'id': '706895394042511360',
+            'title': str,
+            'uploader_id': 'selfisekai',
+            'uploader_url': 'https://selfisekai.tumblr.com/',
+            'age_limit': 0,
+            'tags': [],
+            'like_count': int,
+            'repost_count': int,
+        },
+        'playlist_count': 2,
+    }, {
+        # twitch_live provider - error when linked account is not live
+        'url': 'https://www.tumblr.com/anarcho-skamunist/722224493650722816/hollow-knight-stream-right-now-going-to-fight',
+        'only_matching': True,
     }]
 
     _providers = {
         'instagram': 'Instagram',
         'vimeo': 'Vimeo',
         'vine': 'Vine',
         'youtube': 'Youtube',
+        'dailymotion': 'Dailymotion',
+        'tiktok': 'TikTok',
+        'twitch_live': 'TwitchStream',
+    }
+    # these are known providers, but we don't know which entity type
+    # we are supposed to extract, so we use matching by url
+    _ambiguous_providers = {
+        'bandcamp',
+        'soundcloud',
+    }
```

Comment:
I think we can simplify by moving `_ambiguous_providers` into `_providers` with values of `None`, see below follow-up suggestions